### PR TITLE
Negative Slice Index Support to Bytes.__getitem__

### PR DESCRIPTION
**Pull request by bsthorpe742**
This pull request address Bytes.__getitem__ to support negative indexes when slicing.  Many other objects which support slicing do not support negative indexing at this time.  This could be used as an example to fix other objects which support slicing.  Test cases cover many edge cases with positive and negative step sizes.

### CODE REVIEW

File: tests/datatypes/test_slice.py [hunk 1]
```diff
@@ -75,6 +75,65 @@ def __index__(self):
...
+            print("x[-5:] = ", x[-5:])
+            print("x[:-5] = ", x[:-5])
+            print("x[-2:-8] = ", x[-2:-8])
+
+            print("x[::-2] = ", x[::-2])
+            print("x[-5::-2] = ", x[-5::-2])
+            print("x[:-5:-2] = ", x[:-5:-2])
+            print("x[-2:-8:-2] = ", x[-2:-8:-2])
+            """)
+
```
Comment (simsausaurabh) [inline tests/datatypes/test_slice.py]: Please explain why the testing of negative indexing is done in separate method. I am unable to understand.

Comment (freakboy3742) [inline tests/datatypes/test_slice.py]: I'm not sure what you're referring to. `test_slice_bytes` is a single test method that contains several examples that are tested.
